submissions/antonkottans/js-dom/script.js

```diff
@@ -0,0 +1,136 @@
+    </section>
+    <section>
+        <h3  class="content-heading">Приготовление</h3>
+        Обычно рыба имеет сильный привкус, который не всем по вкусу. Что избавиться от него
+        лучше запекать или тушить.
+    </section>
+</article>`,
+            `
+    <article>
+        <h2  class="content-heading">Сом</h2>
+        <img class="content-img" src="./img/catfish.jpg" alt="catfish">
+        <section>
+            <h3  class="content-heading">Общее описание рыбы</h3>
+            Обитает в реках и озёрах. В передней части тело речного сома имеет округлые формы, а
+            с продвижением к хвосту становится сжатым с обоих боков и переходит в сильный хвост.
+        </section>
+        <section>
+            <h3  class="content-heading">Приготовление</h3>
+            Умеренно жирная рыба. Из-за толстой кожи, можно готовить как угодно и на гриле со
+            сгоревшей в уголь кожей тоже норм, т.к. её далеко не все едят.
+        </section>
+    </article>`,
+            `		<article>
+    <h2  class="content-heading">Карась</h2>
+    <a title="jag asså, Public domain, via Wikimedia Commons" href="https://commons.wikimedia.org/wiki/File:Carassius_carassius_1.jpg"><img class="content-img" alt="Carassius carassius 1" src="https://upload.wikimedia.org/wikipedia/commons/thumb/6/6c/Carassius_carassius_1.jpg/512px-Carassius_carassius_1.jpg"></a>
+    <section>
+        <h3  class="content-heading">Общее описание рыбы</h3>
+        Питаются растительностью, мелкими беспозвоночными, зоопланктоном, зообентосом и
+        детритом. Обитают исключительно в болотистых и низменных озёрах и реках. Спинной
+        плавник длинный, глоточные зубы однорядные. Тело высокое с толстой спиной, умеренно
+        сжатое с боков. Чешуя крупная и гладкая на ощупь.
+    </section>
+    <section>
+        <h3  class="content-heading">Приготовление</h3>
+        Имеет сильный приятный сладковатый вкус и ооочень много костей. В домашних условия
+        подают обычно жареным с крестообразными надрезами. Также очень вкусный получается
+        запеченый с майонезом, сметаной и зеленью.
+    </section>
+</article>`,
+        ];
+        let cashedArticles = [];
+        let fishNames = ["carp", "grassCarp", "pike", "catfish", "carassius"];
+        fishNames.forEach((fishName) =>
+            cashedArticles.push([fishName, document.createElement("div")])
+        );
+        cashedArticles.forEach((contentItem, i) => {
+            contentItem[1].classList.add(contentItem[0], "content");
+            contentItem[1].insertAdjacentHTML("afterbegin", articles[i]);
+        });
+        return cashedArticles;
+    };
+
+    let contentArticles = initializeContentArray();
+
+    let main = document.querySelector(".main");
+    let temporaryFragment = document.createDocumentFragment();
+    contentArticles.forEach((item) => {
+        temporaryFragment.appendChild(item[1]);
+    });
+    main.appendChild(temporaryFragment);
+
```

Comment:
Since you already add all content on page and just hide it with CSS styles, there is no point in using `querySelector` every time someone clicks a button. Just get all elements once and store them in JS.

Reply:
Thus i need to load every time content from js, should i use appendChild every time instead? i use querySelector to hide/show content every time user clicks on button. I didn't get it.

Reply:
oh, do you mean to store nodes?

Reply:
fixed, i guess